information/specification.md

```diff
@@ -0,0 +1,826 @@
+in a transaction, because doing so would incur a fee increase that
+significantly negates the value of including it.
+
+# Interface
+
+All coin selection algorithms used by Cardano Wallet implement a _common
+interface_.
+
+Essentially, a coin selection algorithm is simply a _mathematical function_
+that, when applied to a [standard set of parameters](#parameters), will produce
+a [standard set of results](#results).
+
+This section defines the [parameters](#parameters) accepted by coin selection
+algorithms used in Cardano Wallet, the [results](#results) returned, as well as
+the various [error conditions](#failure-modes) that may occur.
+
+In this section, the terms _coin selection algorithm_ and _coin selection
+function_ will be used interchangeably.
+
+## Parameters
+
+A coin selection function accepts the following parameters:
+
+ 1. #### Requested Output List
+
+    A list of payments to be made to recipient addresses, encoded as a list of
+    [transaction outputs](#transaction-output).
+
+ 2. #### Initial UTxO Set
+
+    A [UTxO set](#utxo-set) from which the coin selection algorithm can select
+    entries, to cover payments listed in the [requested output
+    list](#requested-output-list).
+
+    In the context of a wallet, this parameter would normally be assigned with
+    the wallet's complete [UTxO set](#utxo-set), giving the coin selection
+    algorithm access to the total value associated with that wallet.
+
+ 3. #### Maximum Input Count
+
+    An _upper bound_ on the number of UTxO entries that the coin selection
+    algorithm is permitted to select from the [initial UTxO
+    set](#initial-utxo-set).
+
+## Results
+
+A coin selection function produces the following result values:
+
+ 1. #### Coin Selection
+
+    The _coin selection_ is a record with three fields:
+
+      * A set of **_inputs_**, equivalent to a subset of the
+        [initial UTxO set](#initial-utxo-set).
+
+        Represents the value that has been selected from the wallet in order to
+        cover the total payment value.
+
+      * A set of **_outputs_** (see [transaction output](#transaction-output)).
+
+        Represents the set of payments to be made to recipient addresses.
+
+      * A set of **_change values_**, where each change value is simply a
+        [coin value](#coin-value).
+
+        Represents the change to be returned to the wallet.
+
+ 2. #### Remaining UTxO Set
+
+    The _remaining UTxO set_ is a subset of the [initial UTxO
+    set](#initial-utxo-set).
+
+    It represents the set of values that remain after the coin selection
+    algorithm has removed values to pay for entries in the [requested output
+    list](#requested-output-list).
+
+    In the context of a _wallet_, if a coin selection algorithm is applied to
+    the wallet's _complete_ UTxO set, then the _remaining_ UTxO set represents
+    the _updated_ UTxO set of that wallet.
+
+## Properties
+
```

Comment:
We should probably insert a short paragraph here to explain what this section is about.

Reply:
> We should probably insert a short paragraph here to explain what this section is about.

Fixed in e82a7641abf2c09bdb967a7558e89a6ab196111c.